### PR TITLE
Minor Atomix cleanup

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR removes dead code, primarily an old handler which allowed one to force all partitions to snapshot. This wasn't used anywhere anymore, and was a misnommer since it actually asks partitions to compact (but doesn't force anything) :shrug: 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java [hunk 1]
```diff
@@ -151,62 +144,26 @@ public List<PartitionId> getPartitionIds() {
...
-   *
-   * @return a future to be completed once the snapshot is complete
-   */
-  private CompletableFuture<Void> handleSnapshot() {
-    return Futures.allOf(
-            partitions.values().stream()
-                .map(partition -> partition.snapshot())
-                .collect(Collectors.toList()))
-        .thenApply(v -> null);
+    return ToStringBuilder.reflectionToString(this);
```
Comment (deepthidevaki) [inline atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java:153]: :question: We don't normally use this pattern for toString.

**Review by deepthidevaki — APPROVED** [review]
:rocket: Thanks. Didn't even know about `PartitionEvent`.

File: atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java (hunk 1 above)
Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java:153]: I did see we do, actually, and I liked it better than the ToStringHelper here (though it's mostly on the ZPA team's stuff). But I'm happy to use the usual thing :+1:

**Comment by npepinpe** [comment]
bors merge

**Comment by npepinpe** [comment]
I think I'll backport this just to avoid conflicts with other stuff in the future, and it should be harmless. WDYT?

**Comment by deepthidevaki** [comment]
> I think I'll backport this just to avoid conflicts with other stuff in the future, and it should be harmless. WDYT?

:+1:

File: atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java [hunk 2]
```diff
@@ -151,62 +143,40 @@
...
-            partitions.values().stream()
-                .map(partition -> partition.snapshot())
-                .collect(Collectors.toList()))
-        .thenApply(v -> null);
+    return "RaftPartitionGroup{"
+        + "name='"
+        + name
+        + '\''
+        + ", config="
+        + config
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java:157]: That's very true. It is what we were doing before though, just indirectly (and I guess it couldn't pick it up).

[the remark this answers is not in the document]
Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java:157]: I will just add a toString to the config, that's still useful.

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**